craftai/interpreter.py

```diff
@@ -12,34 +12,43 @@ class Interpreter(object):
   @staticmethod
   def decide(tree, args):
     bare_tree, configuration, tree_version = Interpreter._parse_tree(tree)
+    interpreter = Interpreter._get_interpreter(tree_version)
+
+    return Interpreter._decide(configuration, bare_tree, args, interpreter)
+
+  ####################
+  # Internal helpers #
+  ####################
+
+  @staticmethod
+  def _decide(configuration, bare_tree, args, interpreter):
     if configuration != {}:
       time = None if len(args) == 1 else args[1]
       context_result = Interpreter._rebuild_context(configuration, args[0], time)
       context = context_result["context"]
     else:
       context = Interpreter.join_decide_args(args)
-
     # Convert timezones as integers into standard +/hh:mm format
     # This should only happen when no time generated value is required
     context = Interpreter._convert_timezones_to_standard_format(configuration, context)
 
+    decision = interpreter.decide(configuration, bare_tree, context)
+    decision["context"] = context
+
+    return decision
+
+  @staticmethod
+  def _get_interpreter(tree_version):
     if semver.match(tree_version, ">=1.0.0") and semver.match(tree_version, "<2.0.0"):
-      decision = InterpreterV1.decide(configuration, bare_tree, context)
+      interpreter = InterpreterV1
```

Comment:
```suggestion
      return InterpreterV1
```
I guess you can directly return it here